packages/orchestration/test/network-fakes.ts

```diff
@@ -1,116 +1,274 @@
+const addParamsIfJsonVersion = (
+  version: string,
+  params: { address: string },
+): string => {
+  try {
+    const parsed = JSON.parse(version);
+    return JSON.stringify({
+      ...parsed,
+      ...params,
+    });
+  } catch {
+    return version;
+  }
+};
 
-export const prepareProtocolHandler = (
-  zone: Zone,
-  makeEchoConnectionHandler: MakeEchoConnectionKit,
-  { when },
-) => {
-  const makeProtocolHandler = zone.exoClass(
-    'ProtocolHandler',
-    undefined,
-    () => {
-      return {
-        l: undefined as ListenHandler | undefined,
-        lp: undefined,
-        nonce: 0,
-      };
-    },
-    {
-      async onInstantiate(_port, _localAddr, _remote, _protocol) {
-        return '';
-      },
-      async onCreate(_protocol, _impl) {
-        log('created', _protocol, _impl);
-      },
-      async generatePortID() {
-        this.state.nonce += 1;
-        return `port-${this.state.nonce}`;
+type ImplementedIBCEvents = 'channelOpenAck' | 'acknowledgementPacket';
+
+export const ibcBridgeMocks: {
+  [T in ImplementedIBCEvents]: T extends 'channelOpenAck'
+    ? (
+        obj: IBCMethod<'startChannelOpenInit'>,
+        opts: { bech32Prefix: string; sequence: number },
+      ) => IBCEvent<'channelOpenAck'>
+    : T extends 'acknowledgementPacket'
+      ? (
+          obj: IBCMethod<'sendPacket'>,
+          opts: { sequence: number; acknowledgement: string },
+        ) => IBCEvent<'acknowledgementPacket'>
+      : never;
+} = {
+  channelOpenAck: (
+    obj: IBCMethod<'startChannelOpenInit'>,
+    { bech32Prefix, sequence }: { bech32Prefix: string; sequence: number },
+  ): IBCEvent<'channelOpenAck'> => {
+    const mocklID = Number(obj.packet.source_port.split('-').at(-1));
+    const mockLocalChannelID: IBCChannelID = `channel-${mocklID}`;
+    const mockRemoteChannelID: IBCChannelID = `channel-${mocklID}`;
+    const mockChainAddress =
+      sequence > 0 ? `${bech32Prefix}1test${sequence}` : `${bech32Prefix}1test`;
+
+    return {
+      type: 'IBC_EVENT',
+      blockHeight: 99,
+      blockTime: 1711571357,
+      event: 'channelOpenAck',
+      portID: obj.packet.source_port,
+      channelID: mockLocalChannelID,
+      counterparty: {
+        port_id: obj.packet.destination_port,
+        channel_id: mockRemoteChannelID,
       },
-      async onBind(port, localAddr) {
-        assert(port, `port is supplied to onBind`);
-        assert(localAddr, `local address is supplied to onBind`);
+      counterpartyVersion: addParamsIfJsonVersion(obj.version, {
+        address: mockChainAddress,
+      }),
+      connectionHops: obj.hops,
+      order: obj.order,
+      version: obj.version,
+    };
+  },
+
+  acknowledgementPacket: (
+    obj: IBCMethod<'sendPacket'>,
+    opts: { sequence: number; acknowledgement: string },
+  ): IBCEvent<'acknowledgementPacket'> => {
+    const { sequence, acknowledgement } = opts;
+    return {
+      acknowledgement,
+      blockHeight: 289,
+      blockTime: 1712180320,
```

Comment:
could these be random or incrementing? I don't see any testing of these particular values

Reply:
They could be, but yes, we are not testing anything against `blockTime` or `blockHeight` since application code does not currently rely on them.

These values seems like they could be useful for testing packet timeouts, but I suspect we can cover that path by using `buildMsgErrorString` and implicitly throwing a timeout error through the testing utils.

For testing something like whether `IBCMsgTransferOptions` are honored/correctly specified for `.transfer()`, e2e tests with sim chains seem like a better place for this.